packages/api/src/controllers/stream.ts

```diff
@@ -1327,6 +1327,83 @@ app.patch(
   }
 );
 
+app.post(
+  "/:id/multistream",
+  authorizer({}),
+  validatePost("target-add-payload"),
+  async (req, res) => {
+    const payload = req.body;
+
+    const stream = await db.stream.get(req.params.id);
+
+    if (!stream || stream.deleted) {
+      res.status(404);
+      return res.json({ errors: ["stream not found"] });
+    }
+
+    if (stream.userId !== req.user.id) {
+      res.status(403);
+      return res.json({ errors: ["stream not found"] });
+    }
+    let streamTargets = stream.multistream;
+
+    let multistream: DBStream["multistream"];
+
+    multistream = streamTargets;
+    multistream.targets.push(payload);
```

Comment:
this should work without mutating objects or intermediate duplicate vars
```suggestion
    let multistream: DBStream["multistream"] = { 
      targets: [...(stream.multistream?.targets ?? []), ...payload]
    }
```